numpy/lib/arraypad.py

```diff
@@ -69,8 +69,12 @@ def _linear_ramp(ndim, axis, start, stop, size, reverse=False):
     ramp = ramp.reshape(init_shape)
 
     if size != 0:
+        # In case `stop` < `start` we get a negative gain and need the array
+        # `stop` to have a compatible signed dtype (at least float16)
+        dtype = np.promote_types(stop.dtype, np.float16)
+        stop = stop.astype(dtype)
         # And scale to given start and stop values
-        gain = (stop - start) / float(size)
+        gain = (stop - start) / size
```

Comment:
You might instead want to use `np.lib.histograms._unsigned_subtract`, possibly after moving it to a more sensible shared location

Reply:
Although looking again, that doesn't work too well